# HG changeset patch
# User Rik <rik@octave.org>
# Date 1363140187 25200
#      Tue Mar 12 19:03:07 2013 -0700
# Node ID c22a6cecaeddcdccede33876e6c6b1dfcf6841d6
# Parent  d07aeecb2d229e07fbe526325ef1b42a82ce44d8
build: Use AMCOND to build GUI editor if Qscintilla available.

* configure.ac: Add new AMCOND_HAVE_QSCINTILLA Automake Conditional.
* libgui/src/m-editor/file-editor-interface.h,
libgui/src/m-editor/file-editor-tab.h, libgui/src/m-editor/file-editor.h,
libgui/src/m-editor/find-dialog.h, libgui/src/m-editor/lexer-octave-gui.h:
Remove '#ifdef HAVE_QSCINTILLA' declarations

* libgui/src/module.mk: Check AMCOND_HAVE_QSCINTILLA before adding
GUI editor files to octave_gui_MOC variable.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2629,16 +2629,18 @@ if test $build_gui = yes; then
     else
       ## Let's assume Qscintilla library is at the same location as
       ## other regular Qt libraries.
       QT_LIBS="$QT_LIBS -lqscintilla2"
       OCTAVE_CHECK_FUNC_FINDFIRST_MODERN
       AC_DEFINE(HAVE_QSCINTILLA, 1, 
         [Define to 1 if the QScintilla library and header files are available])
     fi
+    AM_CONDITIONAL([AMCOND_HAVE_QSCINTILLA],
+      [test $octave_cv_lib_qscintilla = yes])
 
     AC_CHECK_FUNCS([setlocale], [],
       [AC_MSG_WARN([setlocale not found -- disabling GUI])
        build_gui=no])
 
     if test $build_gui = yes; then
       case $host_os in
         mingw* | msdosmsvc*)
diff --git a/libgui/src/m-editor/file-editor-interface.h b/libgui/src/m-editor/file-editor-interface.h
--- a/libgui/src/m-editor/file-editor-interface.h
+++ b/libgui/src/m-editor/file-editor-interface.h
@@ -18,18 +18,16 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifndef FILEEDITORINTERFACE_H
 #define FILEEDITORINTERFACE_H
 
-#ifdef HAVE_QSCINTILLA
-
 #include <QMenu>
 #include <QToolBar>
 #include "octave-dock-widget.h"
 
 class file_editor_interface : public octave_dock_widget
 {
   Q_OBJECT
 
@@ -58,11 +56,9 @@ public slots:
 //signals:
 
 //protected:
 
 //protected slots:
 
 };
 
-#endif  // HAVE_QSCINTILLA
-
-#endif  // FILEEDITORINTERFACE_H
+#endif // FILEEDITORINTERFACE_H
diff --git a/libgui/src/m-editor/file-editor-tab.h b/libgui/src/m-editor/file-editor-tab.h
--- a/libgui/src/m-editor/file-editor-tab.h
+++ b/libgui/src/m-editor/file-editor-tab.h
@@ -18,18 +18,16 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifndef FILEEDITORTAB_H
 #define FILEEDITORTAB_H
 
-#ifdef HAVE_QSCINTILLA
-
 #include <Qsci/qsciscintilla.h>
 #include <QWidget>
 #include <QCloseEvent>
 #include <QFileSystemWatcher>
 
 #include "find-dialog.h"
 
 class file_editor;
@@ -153,11 +151,9 @@ private:
 
   QFileSystemWatcher    _file_system_watcher;
 
   find_dialog *         _find_dialog;
   bool                  _find_dialog_is_visible;
   QRect                 _find_dialog_geometry;
 };
 
-#endif  // HAVE_QSCINTILLA
-
-#endif  // FILEEDITORTAB_H
+#endif // FILEEDITORTAB_H
diff --git a/libgui/src/m-editor/file-editor.h b/libgui/src/m-editor/file-editor.h
--- a/libgui/src/m-editor/file-editor.h
+++ b/libgui/src/m-editor/file-editor.h
@@ -18,18 +18,16 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifndef FILEEDITORMDISUBWINDOW_H
 #define FILEEDITORMDISUBWINDOW_H
 
-#ifdef HAVE_QSCINTILLA
-
 #include <QToolBar>
 #include <QAction>
 #include <QMenuBar>
 #include <QStatusBar>
 #include <QCloseEvent>
 #include <QTabWidget>
 
 #include "file-editor-interface.h"
@@ -151,11 +149,9 @@ private:
 
   enum { MaxMRUFiles = 10 };
   QMenu *_mru_file_menu;
   QAction *_mru_file_actions[MaxMRUFiles];
   QStringList _mru_files;
 
 };
 
-#endif  // HAVE_QSCINTILLA
-
-#endif  // FILEEDITORMDISUBWINDOW_H
+#endif // FILEEDITORMDISUBWINDOW_H
diff --git a/libgui/src/m-editor/find-dialog.h b/libgui/src/m-editor/find-dialog.h
--- a/libgui/src/m-editor/find-dialog.h
+++ b/libgui/src/m-editor/find-dialog.h
@@ -37,18 +37,16 @@
 ** Nokia at qt-info@nokia.com.
 ** $QT_END_LICENSE$
 **
 ****************************************************************************/
 
 #ifndef FIND_DIALOG_H
 #define FIND_DIALOG_H
 
-#ifdef HAVE_QSCINTILLA
-
 #include <QDialog>
 #include <Qsci/qsciscintilla.h>
 
 class QCheckBox;
 class QDialogButtonBox;
 class QGroupBox;
 class QLabel;
 class QLineEdit;
@@ -83,12 +81,10 @@ private:
   QPushButton       *_replace_button;
   QPushButton       *_replace_all_button;
   QPushButton       *_more_button;
   QWidget           *_extension;
   QsciScintilla     *_edit_area;
   bool               _find_result_available;
 };
 
-#endif  // HAVE_QSCINTILLA
+#endif // FIND_DIALOG_H
 
-#endif  // FIND_DIALOG_H
-
diff --git a/libgui/src/m-editor/lexer-octave-gui.h b/libgui/src/m-editor/lexer-octave-gui.h
--- a/libgui/src/m-editor/lexer-octave-gui.h
+++ b/libgui/src/m-editor/lexer-octave-gui.h
@@ -18,18 +18,16 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifndef LEXEROCTAVE_H
 #define LEXEROCTAVE_H
 
-#ifdef HAVE_QSCINTILLA
-
 #include "resource-manager.h"
 #include <QObject>
 #include <Qsci/qsciglobal.h>
 #include <Qsci/qscilexer.h>
 #include <Qsci/qsciapis.h>
 
 class lexer_octave_gui : public QsciLexer
 {
@@ -60,11 +58,9 @@ class lexer_octave_gui : public QsciLexe
   QString description (int style) const;
 
 private:
   lexer_octave_gui (const lexer_octave_gui &);
   lexer_octave_gui &operator= (const lexer_octave_gui &);
   QsciAPIs *lexer_api;
 };
 
-#endif  // HAVE_QSCINTILLA
-
-#endif  // LEXEROCTAVE_H
+#endif
diff --git a/libgui/src/module.mk b/libgui/src/module.mk
--- a/libgui/src/module.mk
+++ b/libgui/src/module.mk
@@ -50,22 +50,28 @@ octave_gui_ICONS = \
   src/icons/star.png \
   src/icons/stop.png \
   src/icons/terminal.png \
   src/icons/undo.png \
   src/icons/up.png \
   src/icons/zoom-in.png \
   src/icons/zoom-out.png
 
-octave_gui_MOC = \
+octave_gui_MOC =
+
+if AMCOND_HAVE_QSCINTILLA
+octave_gui_MOC += \
   src/m-editor/moc-file-editor-interface.cc \
   src/m-editor/moc-file-editor-tab.cc \
   src/m-editor/moc-file-editor.cc \
   src/m-editor/moc-find-dialog.cc \
-  src/m-editor/moc-lexer-octave-gui.cc \
+  src/m-editor/moc-lexer-octave-gui.cc
+endif
+
+octave_gui_MOC += \
   src/moc-documentation-dockwidget.cc \
   src/moc-files-dockwidget.cc \
   src/moc-history-dockwidget.cc \
   src/moc-main-window.cc \
   src/moc-octave-qt-event-listener.cc \
   src/moc-settings-dialog.cc \
   src/moc-terminal-dockwidget.cc \
   src/moc-welcome-wizard.cc \
